reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientOperations.java

```diff
@@ -401,6 +411,9 @@ protected void afterInboundComplete() {
 		if (redirecting != null) {
 			listener().onUncaughtException(this, redirecting);
 		}
+		else if (authenticating != null) {
+			listener().onUncaughtException(this, authenticating);
+		}
```

Comment:
It propagates a `HttpClientAuthenticationException` when authentication is required and the request needs to be retried.